lib/spack/spack/repo.py

```diff
@@ -1015,6 +1022,14 @@ def all_packages(self):
         for name in self.all_package_names():
             yield self.get(name)
 
+    def all_package_classes(self):
+        """Iterator over all package *classes* in the repository.
+
+        Use this with care, because loading packages is slow.
```

Comment:
Could this be `@memoized`? Or are you warning about the case of the first call being slow too?

Reply:
Hi @cosmicexplorer and welcome! The message is there to warn about the first call being extremely slow, since it parses all the `package.py` files in the repository and creates Python modules. Everything is then stored in `self._modules` and effectively memoized, see the `_get_pkg_module` implementation for details.